### PR TITLE
Close tabs with middle mouse click

### LINKED ISSUE #2035: [Feature request] Close a workspace tab using middle mouse click
**Synfig version & platform**:
Building from source
Fedora 33

**Issue description**:
Synfig offers multiple tab feature so I'd love if tabs can also be closed using the middle mouse click action just like on Firefox :)

Screenshot from 2021-02-19 14-27-01

### PR DESCRIPTION
**Pull request by Keyikedalube**
Fixes #2035

### CODE REVIEW

File: synfig-studio/src/gui/docks/dockbook.cpp [hunk 1]
```diff
@@ -260,6 +260,13 @@ DockBook::tab_button_pressed(GdkEventButton* event, Dockable* dockable)
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	CanvasView *canvas_view = dynamic_cast<CanvasView*>(dockable);
+
+	// Handle middle mouse click event first before showing the canvas
+	if (event->button == 2) {
```
Comment (ice0) [inline synfig-studio/src/gui/docks/dockbook.cpp:265]: You need to check if `canvas_view` is available after `dynamic_cast`.

```
if (event->button == 2 && canvas_view) {
```

Comment (Keyikedalube) [reply synfig-studio/src/gui/docks/dockbook.cpp:268]: There were two checks done (for middle click and left click) so I thought checking it right after casting would be appropriate https://github.com/synfig/synfig/pull/2107/commits/a2c90d3e36bb370c905e12b436cb9bc6f226db7f

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Comment by rodolforg** [comment]
@Keyikedalube  Ouch I didn't submit it (I didn't even know it would 'saved')

File: synfig-studio/src/gui/docks/dockbook.cpp [hunk 2]
```diff
@@ -260,7 +260,17 @@ DockBook::tab_button_pressed(GdkEventButton* event, Dockable* dockable)
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	CanvasView *canvas_view = dynamic_cast<CanvasView*>(dockable);
-	if (canvas_view && canvas_view != App::get_selected_canvas_view())
+
+	if (!canvas_view)
+		return false;
```
[the remark this answers is not in the document]
Comment (Keyikedalube) [reply synfig-studio/src/gui/docks/dockbook.cpp:265]: Oh
Okay
I'll fix it

**Pull request merged**